parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -91,15 +100,28 @@ customizeK8s() {
 
             retrycmd_if_failure_no_stats 10 15 30 kubectl get cm kubelet-config-1.${CUR_MINOR} -n kube-system -o yaml --kubeconfig ${KUBECONFIG} > kubelet-config.yaml || exit $ERR_ASH_KUBEADM_INIT_JOIN
 
-            sed "/resourceVersion/d" kubelet-config.yaml | sed "s/kubelet-config-1.${CUR_MINOR}/kubelet-config-1.${NEW_MINOR}/1" \
+            sed "/resourceVersion/d" kubelet-config.yaml | sed "s/kubelet-config-1.${CUR_MINOR}/${NEW_KUBELET_CONFIG}/1" \
             | retrycmd_if_failure_no_stats 10 15 30 kubectl apply --kubeconfig ${KUBECONFIG} -f - || exit $ERR_ASH_KUBEADM_INIT_JOIN
 
             retrycmd_if_failure_no_stats 10 15 30 kubectl replace -n kube-system cm kubeadm-conf -f /tmp/kubeadm-upgrade.yaml --kubeconfig ${KUBECONFIG} || exit $ERR_ASH_KUBEADM_INIT_JOIN
 
             retrycmd_if_failure_no_stats 10 15 30 kubectl apply -f /etc/kubernetes/addons/coredns.yaml --kubeconfig ${KUBECONFIG} || exit $ERR_ASH_APPLY_ADDON
+
+        fi
+
+        {{if IsKubernetesVersionGe "1.23.0"}}
+        extractEtcdctl || exit $ERR_ASH_KUBEADM_REFRESH_ETCD_MEMBERLIST
+        masterNodes=$(kubectl get node -o wide --kubeconfig ${KUBECONFIG} | grep -oE "aks-master-\S+")
+        etcdMemberID=$(etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/peer.crt --key /etc/kubernetes/pki/etcd/peer.key --endpoints https://aks-master-{{GetClusterID}}-0:2379,https://aks-master-{{GetClusterID}}-1:2379,https://aks-master-{{GetClusterID}}-2:2379 member list | grep $(hostname) | sed 's/,.*//')
+        if [ -n "${etcdMemberID}" ]; then
+            echo "Removing etcdMember ${etcdMemberID} for $(hostname)"
+            etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/peer.crt --key /etc/kubernetes/pki/etcd/peer.key --endpoints https://aks-master-{{GetClusterID}}-0:2379,https://aks-master-{{GetClusterID}}-1:2379,https://aks-master-{{GetClusterID}}-2:2379 member remove ${etcdMemberID} || exit $ERR_ASH_KUBEADM_REFRESH_ETCD_MEMBERLIST
         fi
+        echo "Removing etcd data directory"
+        rm -rf /var/lib/etcddisk/etcd/
```

Comment:
instead of delete, would it work to move it to a backup directory?

Reply:
We can move to backup directory, /var/lib/etcddisk/bkp